File: HW1/RunLengthEncode.java

public class RunLengthEncode{
    public static String Encode(String text){
        String result = "";
        char current = text.charAt(0);
        int i = 1;
        for(int j = 1; j < text.length();j++){
            if(current == text.charAt(j)){
                i++;
            }else {
                if(i>4){
                    result += '/' + (i < 10 ? "0"+ String.valueOf(i) : String.valueOf(i)) + current;
                }else {
                    for (int j2 = 0; j2 < i; j2++) {
                        result += current;
                    }
                }
                current = text.charAt(j);
                i = 1;
            }

        }
        if(i>4){
            result += '/' + (i < 10 ? "0"+ String.valueOf(i) : String.valueOf(i)) + current;
        }else {
            for (int j2 = 0; j2 < i; j2++) {
                result += current;
            }
        }
        return result;
    }
}

File: HW1/RunLengthEncode.cpp
#include <string>


class RunLengthEncode{
    public:
    static std::string encode(std::string text){
        std::string result = "";
        int i = 1;
        char curr = text[0];
        for (int j = 0; j < text.length(); j++){
            if(curr == text[j])
                i++;
            else{
                if(i > 4){
                    char temp[4];
                    sprintf(temp,"/%.2d%c", i, curr);
                    result += temp;
                }else {
                    result += std::string(i,curr);
                }
                i = 1;
                curr = text[j];
            }
        }
        if(i > 4){
            char temp[4];
            sprintf(temp,"/%.2d%c", i, curr);
            result += temp;
        }else {
            result += std::string(i,curr);
        }
        return result;
    }
};